doc/source/whatsnew/v0.24.0.txt

```diff
@@ -185,6 +185,7 @@ Other Enhancements
 - :class:`Resampler` now is iterable like :class:`GroupBy` (:issue:`15314`).
 - :meth:`Series.resample` and :meth:`DataFrame.resample` have gained the :meth:`Resampler.quantile` (:issue:`15023`).
 - :meth:`Index.to_frame` now supports overriding column name(s) (:issue:`22580`).
+- :class:`ExcelFormatter` now has a `header_style` property. Was previously in global scope (:issue:`22758`).
```

Comment:
Might not actually need this since the ExcelFormatter isn't part of the exposed API

Reply:
Should it be? If we want to support use-cases like https://github.com/pandas-dev/pandas/issues/22758 then it should be. OTOH, I don't think we consider things like HTMLFormatter to be public.

Reply:
That sounds right to me. (and there's nothing stopping people from using it anyway).

Reply:
Opened #22773 to further that discussion; @dannyhyunkim if you can revert the change to whatsnew would be great as this in its current state is more of an internal refactor. Sorry for the back and forth from me on that!